### ISSUE TITLE
lean() function bug in document's field with Buffer type!

### ISSUE THREAD

**Issue by DelavariMohsen** [Automattic/mongoose#7964]
Hi,
I have this model:

```
_id: mongoose.Types.ObjectId,
name: { type: String, required: true },
mimetype: String,
content: Buffer,
```

I return `content` field with this code in Nest js framework:
```
res.contentType(f.mimetype);
res.send(f.content);
```

When I find document with this code:
```
this.files.findById(imageId);
```
It's OK.
but when I use .lean() function in findById Or findOne and return image to client, Image is not valid image file!

I use:
MongoDb cloud v 4.0.
mongoose: ^5.5.7
@nestjs/mongoose: ^6.1.2,

thanks.

**Comment by DelavariMohsen** [comment]
repro script is ready in [my Github](https://github.com/DelavariMohsen/mongoose-lean-buffertype-report-v5.6.4)

**Comment by vkarpov15** [comment]
Unfortunately this is expected behavior and there isn't much Mongoose can do about it because you're using `lean()`. The MongoDB driver stores buffers as [MongoDB's Binary type](http://mongodb.github.io/node-mongodb-native/3.1/api/Binary.html). Mongoose converts MongoDB Binary objects into [Buffers](https://thecodebarbarian.com/an-overview-of-buffers-in-node-js) based on your schema. But we don't do that if you use `lean()`, because the whole point of `lean()` is to avoid schema conversion to minimize performance overhead..

TLDR:

```javascript
    server.get('/getImageWithLean', async (req, res) => {
        const file = await fileModel.findOne({}).lean();
        if (file == null) {
            res.sendStatus(400).send();
            return;
        }
        res.contentType(file.mimetype);
        // Pull out the underlying buffer from the MongoDB Binary instance
        res.send(file.content.buffer);
    });
```
